moveit_core/collision_detection/src/world.cpp

```diff
@@ -136,6 +136,95 @@ bool World::hasObject(const std::string& object_id) const
   return objects_.find(object_id) != objects_.end();
 }
 
+bool World::knowsTransform(const std::string& object_id) const
+{
+  // Check object names first
+  if (objects_.find(object_id) != objects_.end())
+    return true;
+  else  // Then objects' subframes
+  {
+    if (object_id.find("/")!=std::string::npos)
+    {
+      for (const std::pair<std::string, ObjectPtr>& object : objects_)
+      {
+        try 
+        {
+          // Compare both object name and subframe name
+          int name_length = object.second->id_.length();
+          if (object.second->id_ == object_id.substr(0,name_length))
+          {
+            if (object.second->subframe_poses_.find(object_id.substr(name_length+1)) != object.second->subframe_poses_.end())
+              return true;
+          }
+        }
+        catch (const std::exception& e) {;}
+      }
+    }
+  }
+  return false;
+}
+
+const Eigen::Isometry3d& World::getTransform(const std::string& object_id) const
+{
+  // This is almost identical to the function below, except for the return.
```

Comment:
The code duplication is not very nice...

Reply:
I can't remember why I added this comment instead of combining them. If I can't remember tomorrow I'll merge.

Reply:
Done too.